### PR TITLE
Update Go deps

### PR DESCRIPTION
**Pull request by sethvargo**
This fixes a few downstream minor security issues.

### CODE REVIEW

**Comment by myurtoglu** [comment]
Doesn't feel like it's a straightforward change to move from Spring Boot 2 to 3.

**Comment by yolocs** [comment]
Sorry about the conflict

**Comment by sethvargo** [comment]
> Doesn't feel like it's a straightforward change to move from Spring Boot 2 to 3.

I removed the Java changes - just doing Go for now.

> Sorry about the conflict

Not a problem - I moved the protos around a bit, esp talker.

File: internal/talkerpb/talkerpb.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate protoc -I../../third_party/googleapis -I../../internal/protos/talker --go_out=. --go-grpc_out=. --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative talker.proto
+
+package talkerpb
```
Comment (yolocs) [inline internal/talkerpb/talkerpb.go:17]: It probably doesn't have any practical impact but it'd be good if we update https://github.com/abcxyz/lumberjack/blob/7b83e6469d094af974acedb1209ae7377676214f/internal/protos/talker/talker.proto#L18 to be `option go_package = "github.com/abcxyz/lumberjack/internal/talkerpb";`

**Review by yolocs — APPROVED** [review]

**Pull request merged**